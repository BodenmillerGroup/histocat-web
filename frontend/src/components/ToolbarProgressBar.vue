<template>
  <v-progress-linear
    :active="processing"
    :color="color"
    :indeterminate="indeterminate"
    :value="progress"
    class="ma-0"
    height="4"
    style="top: -2px;"
  />
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class ToolbarProgressBar extends Vue {
  @Prop(Boolean) processing?: boolean;
  @Prop(Number) progress?: boolean;
  @Prop(Boolean) indeterminate?: boolean;
  @Prop(String) color?: string;
}
</script>
