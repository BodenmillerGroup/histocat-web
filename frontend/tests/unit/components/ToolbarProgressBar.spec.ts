import { mount } from "@vue/test-utils";
import ToolbarProgressBar from "@/components/ToolbarProgressBar.vue";

import Vue from "vue";
import Vuetify from "vuetify";
Vue.use(Vuetify);

describe("ToolbarProgressBar.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(ToolbarProgressBar);
    expect(wrapper.element).toMatchSnapshot();
  });
});
