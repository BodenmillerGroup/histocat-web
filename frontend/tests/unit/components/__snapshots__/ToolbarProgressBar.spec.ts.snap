// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarProgressBar.vue renders correctly 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow="0"
  class="v-progress-linear ma-0 v-progress-linear--reverse theme--light"
  role="progressbar"
  style="height: 0px; top: -2px;"
>
  <div
    class="v-progress-linear__background primary"
    style="opacity: 0.3; right: 0%; width: 100%;"
  />
  <div
    class="v-progress-linear__buffer"
    style="height: 0px;"
  />
  <transition-stub
    name="slide-x-transition"
  >
    <div
      class="v-progress-linear__determinate primary"
      style="width: 0%;"
    />
  </transition-stub>
</div>
`;
